pkg/kuberneteshelper/pod.go

```diff
@@ -105,3 +106,29 @@ func GetOsmConfigurator(osmNamespace common.MeshNamespace) configurator.Configur
 	cfg := configurator.NewConfigurator(versioned.NewForConfigOrDie(kubeConfig), stop, osmNamespace.String(), constants.OSMMeshConfig)
 	return cfg
 }
+
+// GetMatchingServices returns a list of Kuberentes services in the namespace that match the pod's label
+func GetMatchingServices(kubeClient kubernetes.Interface, pod *v1.Pod) ([]service.MeshService, error) {
```

Comment:
I feel like there should be an existing client go/k8s helper that does this already but I haven't been able to find one, any suggestions are welcome!

Reply:
Pass the pod's labels instead of a pointer to the entire struct.